public/js/controllers/traffic_graphs_controller.js

```diff
@@ -910,157 +883,58 @@ export class TrafficGraphsController {
-                .selectAll("text")
-                .style("text-anchor", "end")
-                .attr("dx", "-.8em")
-                .attr("dy", ".15em")
-                .attr("transform", "rotate(-65)");
-
-            // Add X axis label for the total failed sms graph
-            total_failed_sms_graph
-                .append("text")
-                .attr("class", "redrawElementFailed")
-                .attr(
-                    "transform",
-                    "translate(" + Width / 2 + " ," + (Height + Margin.top + 65) + ")"
-                )
-                .style("text-anchor", "middle")
-                .text("Date (Y-M-D)");
-
-            // Total Sms(s) graph title
-            total_failed_sms_graph
-                .append("text")
-                .attr("class", "redrawElementFailed")
-                .attr("x", Width / 2)
-                .attr("y", 0 - Margin.top / 2)
-                .attr("text-anchor", "middle")
-                .style("font-size", "20px")
-                .style("text-decoration", "bold")
-                .text("Total Failed Message(s) / day");
+            // Tick Values for X axis
+            const tickValuesForXAxis = oneDayFailedChartData.map(d => new Date(d.datetime));
+
+            let oneDayFailedChartConfig = { setFailedMsgGraphTooltipText: true }
+            let xMax = d3.max(oneDayFailedChartData, d => TrafficGraphsController.addOneDayToDate(d.datetime));
+            const oneDayFailedChart = new BarChart(
+                { element: document.querySelector('.total_failed_sms_graph'), data: oneDayFailedChartData });
+            oneDayFailedChart
+                .setTitle("Total Failed Message(s) / day")
+                .setXAxisLabel("Date (Y-M-D)")
+                .setYAxisLabel("No. of Failed Message (s)")
+                .setLegendLabel("total_errored")
+                .setColorScheme("red")
+                .setYLimit(yLimitFailed)
+                .setXAxisTickFormat(dayDateFormatWithWeekdayName)
+                .setTickValuesForXAxis(tickValuesForXAxis)
+                .setXLimitByAddingOneDayDate(xMax)
+                .setGridLinesId("failedGrid")
+                .setBarsRightPadding()
+                .setFactorToShiftBarsToRight()
+                .setConfig(oneDayFailedChartConfig)
+                .draw();
         }
 
         function draw10MinFailedGraph(yLimitFailed) {
+            let _10minDayFailedChartData = JSON.parse(JSON.stringify(dataFilteredWeek));
+            _10minDayFailedChartData.forEach(function(d) {
+                d.datetime = new Date(d.datetime);
+                d.value = +d.total_errored;
+            })  
             // Set Y axis limit to max of daily values or to the value inputted by the user
             if (isYLimitFailedManuallySet == false) {
-                yLimitFailed = d3.max(dataFilteredWeek, d => d.total_errored);
+                yLimitFailed = d3.max(_10minDayFailedChartData, d => d.value);
```

Comment:
is this 10MinDayFailedData  or just 10MinFailedChartData?

Reply:
10MinFailedChartData

Reply:
I've updated the variable to _10minFailedChartData